src/app/services/campaign.service.ts

```diff
@@ -482,9 +482,9 @@ export class CampaignService {
     self.methods.anyKnownNPCs = (): boolean => self.methods.listKnownNPCs().length > 0
     self.methods.isActiveNPC = (npc: CampaignNpc): boolean => npc.id === self.locals.active_npc
 
-    self.methods.addNPC = (npc: CampaignNpc = new CampaignNpc): void => {
+    self.methods.addNPC = (npc: CampaignNpc = new CampaignNpc, name: string): void => {
       npc.pos = self.methods.listNPCs().length
-      npc.name = `${this.sheetSvc.randomName()} ${this.sheetSvc.randomName()}`
+      npc.name = name ? name : `${this.sheetSvc.randomName()} ${this.sheetSvc.randomName()}`
```

Comment:
```suggestion
      npc.name = npc.name ? `Copy of ${npc.name}` : `${this.sheetSvc.randomName()} ${this.sheetSvc.randomName()}`
```

Reply:
Hey we will have issues when moving NPCs, the moved NPC will be concatenated with 'copy of'.

Reply:
I added the renaming of the npc on Clone NPC method then!